### PR TITLE
DEX-138 test orderbook deletion

### PR DESCRIPTION
**Pull request by msmolyakov**

### CODE REVIEW

File: it/src/test/scala/com/wavesplatform/it/sync/matcher/MatcherTestSuite.scala [hunk 1]
```diff
@@ -81,7 +82,7 @@ class MatcherTestSuite extends FreeSpec with Matchers with BeforeAndAfterAll wit
         // Alice check that order is correct
         val orders = matcherNode.orderBook(aliceWavesPair)
         orders.asks.head.amount shouldBe aliceSellAmount
-        orders.asks.head.price shouldBe 2.waves * Order.PriceConstant
+        orders.asks.head.price shouldBe 2.waves * PriceConstant
```
Comment (DrBlast) [inline it/src/test/scala/com/wavesplatform/it/sync/matcher/MatcherTestSuite.scala]: Delete PriceConstant  from all matcher testsuites

File: it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
+package com.wavesplatform.it.sync.matcher
+
+import com.typesafe.config.Config
+import com.wavesplatform.it.ReportingTestName
+import com.wavesplatform.it.api.SyncHttpApi._
+import com.wavesplatform.it.api.SyncMatcherHttpApi._
+import com.wavesplatform.it.sync.matcher.config.MatcherDefaultConfig._
```
Comment (DrBlast) [inline it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala]: Use MatcherPiceAssetConfig. In test jsut issue assets. All pairs already created in config

File: it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      matcherNode.orderStatus(sellOrder, aliceWavesPair).status shouldBe "PartiallyFilled"
+      matcherNode.orderStatus(anotherSellOrder, aliceWavesPair).status shouldBe "Accepted"
+      matcherNode.orderStatus(buyOrder, aliceWavesPair).status shouldBe "Accepted"
+      matcherNode.orderStatus(orderForAnotherPair, aliceWavesPair).status shouldBe "Accepted"
+
+      matcherNode.reservedBalance(aliceNode)(aliceAsset) should be > 0L
+      matcherNode.reservedBalance(bobNode)("WAVES") should be > 0L
+      matcherNode.reservedBalance(aliceNode)(aliceAsset2) should be > 0L
+
+      matcherNode.deleteOrderBook(aliceWavesPair)
```
Comment (DrBlast) [inline it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala]: Split test in three parts: preparation, orderbook deletion and orderbook del not affects another pair with this asset.
wctUsd and wctWaves pairs are the best choise for this test

File: it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+    val sellOrder = matcherNode.placeOrder(bobNode, wctUsdPair, SELL, amount, 2 * price).message.id
+
+    matcherNode.waitOrderStatus(wctUsdPair, buyOrder, "PartiallyFilled")
+    matcherNode.waitOrderStatus(wctUsdPair, submitted, "Filled")
+
+    val (aliceReservesForOnePair, bobReservesForOnePair) = (reservesOf(aliceNode), reservesOf(bobNode))
+
+    val buyOrderForAnotherPair  = matcherNode.placeOrder(aliceNode, wctWavesPair, BUY, amount, price).message.id
+    val sellOrderForAnotherPair = matcherNode.placeOrder(bobNode, wctWavesPair, SELL, amount, 2 * price).message.id
```
Comment (DrBlast) [inline it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala]: one order is enought for this case

File: it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+    matcherNode.waitOrderStatus(wctWavesPair, buyOrderForAnotherPair, "Accepted")
+    matcherNode.waitOrderStatus(wctWavesPair, sellOrderForAnotherPair, "Accepted")
+
+    val (aliceReservesForBothPairs, bobReservesForBothPairs) = (reservesOf(aliceNode), reservesOf(bobNode))
+
+    matcherNode.deleteOrderBook(wctUsdPair)
+
+    "orders by the pair should be canceled" in {
```
Comment (DrBlast) [inline it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala]: add "after delete" to the test name

File: it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+      matcherNode.waitOrderStatus(wctUsdPair, buyOrder, "Cancelled")
+      matcherNode.waitOrderStatus(wctUsdPair, anotherBuyOrder, "Cancelled")
+      matcherNode.waitOrderStatus(wctUsdPair, sellOrder, "Cancelled")
+    }
+
+    "orderbook was really deleted" in {
```
Comment (DrBlast) [inline it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala]: add check for market status

File: it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala (hunk 4 above)
Comment (msmolyakov) [reply it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala]: both orders will remain for filling bids/asks

File: it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala (hunk 5 above)
Comment (msmolyakov) [reply it/src/test/scala/com/wavesplatform/it/sync/matcher/OrderBookTestSuite.scala]: It's describe in the parent scope "When delete order book": https://github.com/wavesplatform/Waves/pull/1589/files#diff-18bfca4a15099c163a413342145e7d4aR33

**Pull request merged**